src/overlays/actors/ovl_Bg_Spot15_Rrbox/z_bg_spot15_rrbox.c

```diff
@@ -21,45 +39,343 @@ const ActorInit Bg_Spot15_Rrbox_InitVars = {
     (ActorFunc)BgSpot15Rrbox_Update,
     (ActorFunc)BgSpot15Rrbox_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B3960.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B39E8.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+Vec3f D_808B45C4[] = {
+    { 770.0f, 1490.0f, -299.0f },
+    { 770.0f, 1550.0f, -299.0f },
+};
+
+Vec3f D_808B45DC[] = { { 29.99f, 0.01f, -29.99f },
+                       { -29.99f, 0.01f, -29.99f },
+                       { -29.99f, 0.01f, 29.99f },
+                       { 29.99f, 0.01f, 29.99f },
+                       { 0.0f, 0.01f, 0.0f } };
+
+s32 D_808B4618[] = { 0, 0 };
+
+void func_808B3960(BgSpot15Rrbox* this, GlobalContext* globalCtx, s32* arg2, s32 flags) {
+    s32 temp1 = 0xAB;
+    s32 sp30 = 0;
+    u32 temp2 = 0x32;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, flags);
+    DynaPolyInfo_Alloc(arg2, &sp30);
+
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp30);
+
+    if (this->dyna.dynaPolyId == temp2) {
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_spot15_rrbox.c", temp1,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void func_808B39E8(Vec3f* arg0, Vec3f* arg1, f32 arg2, f32 arg3) {
+    arg0->x = ((arg1->z * arg2) + (arg1->x * arg3));
+    arg0->y = arg1->y;
+    arg0->z = ((arg1->z * arg3) - (arg1->x * arg2));
+}
+
+void func_808B3A34(BgSpot15Rrbox* this) {
+    this->bgId = 50;
+}
+
+s32 func_808B3A40(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    DynaPolyActor* temp_v0;
+
+    temp_v0 = DynaPolyInfo_GetActor(&globalCtx->colCtx, this->bgId);
+
+    if (temp_v0 != 0) {
+        if (Math3D_Dist2DSq(temp_v0->actor.posRot.pos.x, temp_v0->actor.posRot.pos.z, this->dyna.actor.posRot.pos.x,
+                            this->dyna.actor.posRot.pos.z) < 0.01f) {
+            return true;
+        }
+    }
+    return false;
+}
+
+s32 func_808B3AAC(BgSpot15Rrbox* this, GlobalContext* globalCxt) {
+    s16 phi_v0;
+    Actor* actor = &this->dyna.actor;
+
+    if (globalCxt->sceneNum == 0x4C) {
+        return 1;
+    } else if (func_808B3A40(this, globalCxt) != 0) {
+        return 0;
+    }
+
+    if (actor->posRot.pos.x <= 930.0f && -360.0f <= actor->posRot.pos.z) {
+        if (0.0f <= this->dyna.unk_150) {
+            phi_v0 = actor->posRot.rot.y;
+        } else {
+            phi_v0 = actor->posRot.rot.y + 0x8000;
+        }
+
+        if (phi_v0 < 0x2000 && phi_v0 >= -0x5FFF) {
+            return gSaveContext.eventChkInf[1] & 0x10;
+        }
+        return 1;
```

Comment:
is this return here required to match? looks like it can be removed

Reply:
It appears to actually be required.